src/check_pack.c

```diff
@@ -196,7 +202,11 @@ static char *upack_str(char **buf)
     char *val;
     int strsz;
 
-    strsz = upack_int(buf);
+    ck_uint32 str_sz = upack_int(buf);
+    if(str_sz > INT_MAX)
```

Comment:
The check for the pack and unpack are not the same. A check for UINT32_MAX is done for the pack, but a check for INT_MAX is done for the unpack. Should the check be the same, and use the smaller of UINT32_MAX or INT_MAX?

Reply:
I see the point looking at some of the other commits that do the same thing. It is not about catching an issue when the value is first packed, but doing a sanity check before the value is used. That should be fine. It catches an issue before the data is used improperly.

Reply:
Yes. It checks that the value is not too big. In fact, I am unhappy about writing a ck_uint32 to an int, because int's size could be smaller (and system dependent). In fact, if int is 32 bit, it is certainly too small because of the unsigned to signed conversion.
But DurationMsg is part of the API (the public header). If that were changed, it would remove the need for many checks.